File: contents/contents/bogo_sort/code/javascript/bogo.js
function isSorted(arr) {
  for (let i = 0; i < arr.length - 1; i++) {
    if (arr[i] > arr[i + 1]) {
      return false;
    }
  }

  return true;
}

function bogoSort(arr) {
  while (!isSorted(arr)) {
    shuffle(arr);
  }
}

function shuffle(arr) {
  for (let r = arr.length -1; r > 0; r--) {
    let i = Math.floor(Math.random() * r);
    let tmp = arr[i];
    arr[i] = arr[r];
    arr[r] = tmp;
  }
}

function main() {
  const testArray = [4, 5, 123, 24, 34, -5];
  bogoSort(testArray);
  console.log(testArray);
}

main();

File: contents/contents/bogo_sort/code/python/bogo.py
import random


def is_sorted(a):
    for i in range(len(a)-1):
        if a[i+1] < a[i]:
            return False
    return True

def bogo_sort(a):
    while not is_sorted(a):
        random.shuffle(a)

def main():
    a = [1, 3, 2, 4]
    bogo_sort(a)
    print(a)

main()

